torch_frame/nn/encoder/stype_encoder.py

```diff
@@ -685,3 +699,67 @@ def encode_forward(self, feat: TensorData) -> Tensor:
         # -> [batch_size, num_cols, out_channels]
         x = x_lin + self.bias
         return x
+
+
+class Time2VecEncoder(StypeEncoder):
+    r"""Time2VecEncoder for encoding timestamp stype. The input tensor is
+    expected to have 7 columns representing year, month, day, day of week,
+    hour, minute, and second. The encoding includes both a linear component
+    (for the year) and periodic components (for month, day, day of week,
+    hour, minute, second). The linear component applies
+    :class:`torch.nn.Linear` on the year and the periodic components apply
+    sin function on the rest of the columns. And then a
+    :class:`torch.nn.Linear` is applied to the concatenated components.
```

Comment:
Don't get that. Why not apply periodic components on subset, merge with year, and then apply a single `Linear` layer?